### PR TITLE
Implement offset and @ modifiers for vector/matrix selectors

### LINKED ISSUE #91: [tsdb] evaluate vector selectors with respect to the specified offset and at modifiers
## Summary

PromQL allows specifying `offset` and `@` modifiers when selecting vectors. The evaluator currently just ignores these. We need to extend vector selection to adjust the start/end times of the vector lookups based on these modifiers.

## Motivation

Full PromQL support

## Proposed Solution

See above

## Additional Context

https://prometheus.io/docs/prometheus/latest/querying/basics/#offset-modifier

### PR DESCRIPTION
**Pull request by samsond**
## Summary
Adds support for PromQL time modifiers to achieve Prometheus compatibility:
- `offset <duration>`: shifts evaluation time backward (positive) or forward (negative)
- `@ <timestamp>`: sets absolute evaluation time
- `@ start()` / `@ end()`: uses query envelope boundaries

## Implementation details:
- Added `apply_time_modifiers()` helper that applies `@` first, then `offset`
- Updated `evaluate_vector_selector()` and `evaluate_matrix_selector()` to accept three time parameters: `evaluation_ts` (per-step time), `query_start`, and `query_end` (query envelope for `@ start()`/`@ end()`)
- Clarified time parameter roles by introducing `evaluation_ts` for per-step evaluation time (avoiding overloading `end`)
- For instant queries, all three time parameters equal the query time

## Related Issues

Fixes #91 

## Test Plan

Added 5 new tests covering:

* offset
* @ <timestamp> 
* @ start() / @ end()
* combined @ + offset semantics

All existing and new tests pass (401 total)

## Checklist

- [X] Tests added/updated
- [x] `cargo fmt` and `cargo clippy` pass
- [ ] Documentation updated (if applicable)

### CODE REVIEW

File: timeseries/src/promql/evaluator.rs [hunk 1]
```diff
@@ -2294,4 +2440,413 @@ mod tests {
             );
         }
     }
+
+    #[tokio::test]
+    async fn should_evaluate_vector_selector_with_offset_modifier() {
```
Comment (agavra) [inline timeseries/src/promql/evaluator.rs]: can we add tests for evaluating with positive and negative offsets? and it would be good to have tests for timestamps that don't align exactly with the evaluation timestamp.

File: timeseries/src/promql/evaluator.rs [hunk 2]
```diff
@@ -611,6 +664,59 @@ impl<'reader, R: QueryReader> Evaluator<'reader, R> {
...
+                AtModifier::Start => query_start,
+                AtModifier::End => query_end,
+            };
+        }
+
+        // Apply offset modifier (relative adjustment)
+        if let Some(offset) = &vector_selector.offset {
+            adjusted_time = match offset {
+                Offset::Pos(duration) => {
+                    // Positive offset: look back in time
```
Comment (agavra) [inline timeseries/src/promql/evaluator.rs]: nit: since this seems counterintuitive let's just make it extra clear that this is intentional by giving an example and noting this is how prometheus works (I double checked)

File: timeseries/src/promql/evaluator.rs (hunk 1 above)
Comment (samsond) [reply timeseries/src/promql/evaluator.rs]: Agreed, I will add coverage.

File: timeseries/src/promql/evaluator.rs (hunk 2 above)
Comment (samsond) [reply timeseries/src/promql/evaluator.rs]: Good point. I will update.

**Comment by agavra** [comment]
Merging. Thanks again @samsond for the contribution 😄
